libs/java/auth_core/src/main/java/com/yahoo/athenz/auth/token/Token.java

```diff
@@ -133,7 +133,7 @@ public boolean validate(PublicKey publicKey, int allowedOffset, boolean allowNoE
         // explicitly enabled that option for our system. by default
         // they should have an expiration date of less than 30 days
         
-        if (expiryTime != 0 || !ATHENZ_TOKEN_NO_EXPIRY || !allowNoExpiry) {
+        if (expiryTime != 0 && !ATHENZ_TOKEN_NO_EXPIRY && !allowNoExpiry) {
```

Comment:
We cannot change ||'s to &&. That would introduce a security issue where the library will no longer validate expiry times. That's not the job of subclasses. The implementation requires that if an expiry time is specified (regardless of what other options are configured), then it must be validated. If you make those ||'s to &&'s then if I specify an expired token with the option that the caller is allowed to accept tokens with no expiry, then the token will be returned as valid. This method is used by all authorities, etc. So it's not just in use by ZPE. So the correct implementation is correct and will not be changed.

Reply:
I was thinking of fixing each authority coinciding with this opportunity, but after seeing your proposal, I've decided to leave it as is.